### ISSUE TITLE
Delete images/frames from an existing task

### ISSUE THREAD

**Issue by carlescarmonacalpe** [opencv/cvat#95]
Hi,

Thank you for this amazing tools, seems really promising. I'd like to suggest a new feature which is to allow to add/delete images/videos from an existing task. From my side it's really common to extend or reduce the dataset size during the development process.

Thank you for you work!

**Comment by carlescarmonacalpe** [comment]
Hi, 

The only advantage is that you don't need to modify/join the exported files If you annotate two different tasks and you need to have both annotations together to train a model.

**Comment by nmanovic** [comment]
Hi @carlescarmonacalpe ,

If you have a good idea how to implement that please send us a PR. Our internal focus is to add stability, introduce analytics, improve UI and implement a couple of semi-automated methods to annotate data. Please let me close the issue for now but if you like implement such functionality don't hesitate to reopen it. In any case I really appreciate your involvements. Thank you very much!

**Comment by nmanovic** [comment]
Probably it isn't a bad idea to treat an annotation task like a dataset (which can be extended or reduced). It is especially good if we will be able to change status of an annotation task automatically (e.g. done -> annotate). Also a couple of internal customers also requested the feature.

**Comment by nmanovic** [comment]
> Vitor Casadei 04:41
Hi, I used to use vatic and it exported the videos in image format. However on CVAT I can only download the labels, is that right? I've got to convert frames to images on my own?

> vcasadei , at the moment you need to prepare data (at least for videos) by yourself. Internally we don't keep original data to save server disk space. When you upload data CVAT does compress images (image_quality parameter when you create an annotation task). Thus 1 frame can weight only 10-100 kilobytes. For example, an original image in png format can be several megabytes. I will submit an issue regarding that. Probably we need one more step in CVAT: prepare a dataset. Thus users can prepare data and after that download them from CVAT as is. If you can help us with such app/feature for CVAT you are welcome.

**Comment by nmanovic** [comment]
The feature will be implemented only in our new UI. Thus need to move it on v1.0.0 which is expected at the end of the year.

**Comment by pagujeeva** [comment]
I am facing similar issue where I will have to consolidate/add additional frames with annotation and generate one single JSON file. Is there a way, I can do that?

**Comment by illuminati945** [comment]
Hi,
Is there a progress update on this feature?

**Comment by michellemul** [comment]
Hello! Is there anything more with this? I have an extra 40 images to annotate so was hoping to just add them into my existing annotations. Cheers!

**Comment by mrawding** [comment]
Hi, any update on this feature? Would be extremely useful !! CVAT is great, but this definitely helps users who might have made mistakes when uploading datasets and didn't realize until after annotation. Furthermore, it is beneficial to any user who would like to continue to add to their datasets as they annotate. It makes exportation of annotations easier and more consolidated when going to train.

**Comment by nmanovic** [comment]
@mrawding , the PR is the first step to solve the problem https://github.com/openvinotoolkit/cvat/pull/2255

**Comment by davodogster** [comment]
Hello @nmanovic and @bsekachev Does it work now? Adding images to an existing task, or merging tasks within a project.

**Comment by bsekachev** [comment]
@davodogster

No updates on the feature.
Instead of adding images to an existing task, you can move the task to a project and create another task in the project with new images. Finally, you can dump all the annotations for the project, after #3365  will be merged. 

As a workaround, you can add to extra images a tag: "extra" for example. And if the annotation file contains this tag, you just ignore the image.

**Comment by davodogster** [comment]
@bsekachev Awesome, I'm looking forward to using this feature when it has been merged. Thank you

**Comment by nmanovic** [comment]
@davodogster , it was merged already. Please use the latest development version if you want to test/use the feature.

To close the issue we are waiting ability to mark images/frames as removed in a task.

**Comment by davodogster** [comment]
@nmanovic Great, thanks. So I need to docker-compose stop, backup my annotations and then clone the latest cvat github repository and then docker-compose again ? Regards, Sam

**Comment by nmanovic** [comment]
@davodogster , it is the fastest way.

**Comment by davodogster** [comment]
@nmanovic Hi guys, I just cloned the latest version of cvat and ran docker compose and modified the docker compose file (I notice there were some big changes to the docker compose file) see here:

![image](https://user-images.githubusercontent.com/46079516/129834211-7ef3c7a9-5c82-494c-96eb-62ac7e0f3a94.png)

It deployed successfully but there is no option for exporting a project as a dataset in the "Actions" tab.

![image](https://user-images.githubusercontent.com/46079516/129834633-16b0a805-69f3-4b77-94f8-247e61c0bec0.png)

Am I doing something wrong?

**Comment by azhavoro** [comment]
> @nmanovic Hi guys, I just cloned the latest version of cvat and ran docker compose and modified the docker compose file (I notice there were some big changes to the docker compose file) see here:
> 
> ![image](https://user-images.githubusercontent.com/46079516/129834211-7ef3c7a9-5c82-494c-96eb-62ac7e0f3a94.png)
> 
> It deployed successfully but there is no option for exporting a project as a dataset in the "Actions" tab.
> 
> ![image](https://user-images.githubusercontent.com/46079516/129834633-16b0a805-69f3-4b77-94f8-247e61c0bec0.png)
> 
> Am I doing something wrong?

Hi, please use `Export task dataset` option in the `Actions` tab.

Regarding orphans container - it's expected, just follow the docker suggestion and add `--remove-orphans` once.

**Comment by davodogster** [comment]
Hi,

That is not an option. The only option is delete. That is why I asked the question.

Cheers, Sam

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ryanalexmartin** [comment]
The workaround described above, exporting the labeled data and using those labels on a subset of images, does not work when using COCO annotations, if the images you wish to ignore already contain labels.  I don't think the workaround provided above is sufficient for most CVAT users.  This is a rather basic functionality, it is perhaps my team's greatest frustration with CVAT.

Is it possible to exec into the PostgreSQL DB and delete annotated images from there?

**Comment by domef** [comment]
any update on this feature?

**Comment by bsekachev** [comment]
Hi guys, @carlescarmonacalpe @domef 

The feature has been finished and implemented in our CVAT fork, here: https://github.com/cvat-ai/cvat

**Comment by adamsau** [comment]
+1 really need this feature

**Comment by nmanovic** [comment]
@adamsau , is it something which you are looking for? https://cvat-ai.github.io/cvat/docs/manual/advanced/delete_frame/

![](https://cvat-ai.github.io/cvat/images/image245.jpg)

**Comment by adamsau** [comment]
yes thanks a lot!

**Comment by bsekachev** [comment]
It is possible now, you can update CVAT from develop branch if you want to use the feature.

**Comment by valentin-fontanger** [comment]
Can somebody confirm this : when I delete a frame that has been annotated on the UI, and I export the dataset, this frame is still present in my exported annotation and image folder.

**Comment by ooopus** [comment]
> Can somebody confirm this : when I delete a frame that has been annotated on the UI, and I export the dataset, this frame is still present in my exported annotation and image folder.

I encountered the same problem.

**Comment by ooopus** [comment]
> Can somebody confirm this : when I delete a frame that has been annotated on the UI, and I export the dataset, this frame is still present in my exported annotation and image folder.

I found a fix: exec into the `cvat_server` container and remove the files in `data/cache/export`.
